packages/agent/src/agent/CodexSDKAgent.ts

```diff
@@ -0,0 +1,429 @@
+/**
+ * @license
+ * Copyright 2025 BrowserOS
+ */
+
+import { Codex, type McpServerConfig } from '@browseros/codex-sdk-ts'
+import { EventFormatter, FormattedEvent } from '../utils/EventFormatter.js'
+import { logger, fetchBrowserOSConfig, type BrowserOSConfig } from '@browseros/common'
+import type { AgentConfig } from './types.js'
+import { BaseAgent } from './BaseAgent.js'
+import { AGENT_SYSTEM_PROMPT } from './Agent.prompt.js'
+import { allControllerTools } from '@browseros/tools/controller-based'
+import { ControllerBridge } from '@browseros/controller-server'
+
+/**
+ * Environment variable configuration
+ */
+const ENV = {
+  CODEX_BINARY_PATH: process.env.CODEX_BINARY_PATH || '/opt/homebrew/bin/codex',
+  MCP_SERVER_HOST: process.env.MCP_SERVER_HOST || '127.0.0.1',
+  MCP_SERVER_PORT: process.env.HTTP_MCP_PORT || '9100',
+  BROWSEROS_CONFIG_URL: process.env.BROWSEROS_CONFIG_URL,
+  OPENAI_API_KEY: process.env.OPENAI_API_KEY
+} as const
+
```

Comment:
Other than CODEX_BINARY_PATH everything else can in agent cofig right? and they will be dynamic. Let's refactor the `server` -> `agent` with a clean interface that is extensible.